Add connect with SSL/TLS

marcopellin opened the pull request:
Added an overloaded function for doConnect with SSL/TLS certificates.
Added QTDIR in CMakeLists.txt.

marcopellin committed:
Add connect with SSL/TLS

skyhisi commented on README.md:
```diff
@@ -5,7 +5,7 @@ Wrapper of the [Mosquitto](http://mosquitto.org/) library for [Qt](http://www.qt
 ## Building on Linux
     cd build
-    cmake -D CMAKE_BUILD_TYPE=Debug ..\source
+    cmake -D CMAKE_BUILD_TYPE=Debug -D QTDIR=/your_home/Qt/5.x/gcc_64/lib/cmake ..\source
```
This isn't required if the distro provides QT 5, so it shouldn't be in the default instructions.

You could have a "Building on Linux with different QTDIR" section.

skyhisi commented on source/qtmosquitto.cpp:
```diff
@@ -103,6 +103,38 @@ bool QtMosquittoClient::doConnect(const QString& host, int port, int keepalive)
+      return false;
+    }
+
+    // Connect
```
This is duplicate code.

It would be better to just add a separate function "configureTls" that you call before the "doConnect".

marcopellin replied:
That's a good idea, thank you.

marcopellin committed:
Configure SSL/TLS

marcopellin committed:
Update README.md

marcopellin committed:
Add SSL/TLS advanced options

marcopellin committed:
Add build with qmake

marcopellin committed:
Improved build with qmake

marcopellin committed:
Fixed README

marcopellin committed:
Set max_inflight_messages

marcopellin committed:
Create LICENSE

marco-pellin-enetec committed:
revert last commit

marcopellin committed:
Create LICENSE

Add EPL 1.0 as suggested by the main contributor of the project:

/*
Copyright (c) 2015 Silas Parker <skyhisi@gmail.com>
All rights reserved. This program and the accompanying materials
are made available under the terms of the Eclipse Public License v1.0
and Eclipse Distribution License v1.0 which accompany this distribution.
 
The Eclipse Public License is available at
   http://www.eclipse.org/legal/epl-v10.html
and the Eclipse Distribution License is available at
  http://www.eclipse.org/org/documents/edl-v10.php.
 
Contributors:
   Silas Parker
*/

marcopellin committed:
Update Travis CI config 

From main contributor

marcopellin committed:
Update README

Edit travis icon path

skyhisi commented:
Hi Marco,
Sorry it took me so long to getting around to merging this, I haven't really used this library since I posted it.
Thanks,
Silas
